src/main/java/com/squareup/javapoet/ClassName.java

```diff
@@ -122,7 +122,7 @@ public static ClassName bestGuess(String classNameString) {
 
     // Add the package name, like "java.util.concurrent", or "" for no package.
     int p = 0;
-    while (p < classNameString.length() && Util.isLowerCase(classNameString.charAt(p))) {
+    while (p < classNameString.length() && Character.isLowerCase(classNameString.charAt(p))) {
```

Comment:
Strictly speaking this should be classNameString.codePointAt(p). There are actually letters in the Supplementary Multilingual Plane, for example [Mathematical Bold Capital A](http://www.fileformat.info/info/unicode/char/1d400/index.htm). It's not very likely you're going to encounter them here, but on the other hand it doesn't cost anything to use the codePoint variants rather than the char ones.